_resources/port1.0/group/golang-1.0.tcl

```diff
@@ -227,6 +227,10 @@ proc handle_set_go_vendors {vendors_str} {
                         set distfile ${vproject}-${vversion}.tar.gz
                         set master_site https://gitlab.com/${vauthor}/${vproject}/-/archive/${vversion}
                     }
+                    salsa.debian.org {
+                        set distfile ${vproject}-${vversion}.tar.gz
+                        set master_site https://salsa.debian.org/${vauthor}/${vproject}/-/archive/${vversion}
+                    }
```

Comment:
We should not duplicate the code handling GitLab. Fortunately, Tcl's [switch case](https://www.tcl.tk/man/tcl8.5/TclCmd/switch.htm) supports a fallthrough option for this purpose.

Replacing both the gitlab.com and the salsa.debian.org block in your code with

```suggestion
                    gitlab.com -
                    salsa.debian.org {
                        set distfile ${vproject}-${vversion}.tar.gz
                        set master_site https://${vdomain}/${vauthor}/${vproject}/-/archive/${vversion}
                    }
```

should achieve this.